### PR TITLE
feat: add props zenModeEnabled and gridModeEnabled so host can control completely

### LINKED ISSUE #2882: Allow host to control zenmode / gridmode completely
Right now we allow `zenMode` , `gridMode` to be passed via `initialData.appState` but thats only for initial load.
We should allow complete control to the host as well which means a controlled component.

**Use case**
The host can have some action that triggers these modes or if the host always wants to have a UI with any of these modes.

**Solution**
We can allow prop for `zenMode`, `gridMode`, and any such actions and if these props are true we shouldn't render the options in the context menu or buttons to exit these modes as well hence fully controlled by the host. 
And this will also avoid confusion in our codebase of when to consider host prop.

cc @dwelle

### PR DESCRIPTION
**Pull request by ad1992**
fixes #2882 
check it [here](https://codesandbox.io/s/excalidraw-v3-wcqdl)

TODOS
- [x] update readme
- [x] update changelog

### CODE REVIEW

**Review by dwelle — APPROVED** [review]

**Comment by dwelle** [comment]
Maybe write a couple of tests coz I have a feeling this will regress :)

File: src/packages/excalidraw/README.md [hunk 1]
```diff
@@ -335,3 +337,11 @@ A function that renders (returns JSX) custom UI footer. For example, you can use
...
 
 This prop indicates if the app is in `view mode`. When this prop is used, the `view mode` will not show up in context menu is so it is fully controlled by host. Also the value of this prop if passed will be used over the value of `intialData.appState.viewModeEnabled`
+
+#### `zenModeEnabled`
+
+This prop indicates if zen mode is enabled. When this prop is used, the `zen mode` will not show up in context menu is so it is fully controlled by host. The `Exit zen mode` button will also not show up if this prop is true. Also the value of this prop if passed will be used over the value of `intialData.appState.zenModeEnabled`
+
+#### `gridModeEnabled`
+
+This prop indicates if grid mode is enabled. When this prop is used, the `grid mode` will not show up in context menu is so it is fully controlled by host. Also the value of this prop if passed will be used over the value of `intialData.appState.gridModeEnabled`
```
Comment (ad1992) [inline src/packages/excalidraw/README.md:347]:
```suggestion
This prop indicates if `grid mode` is enabled. When this prop is used, the `grid mode` will not show up in context menu is so it is fully controlled by host. Also the value of this prop if passed will be used over the value of `intialData.appState.gridModeEnabled`
```

File: src/packages/excalidraw/README.md [hunk 2]
```diff
@@ -335,3 +337,11 @@ A function that renders (returns JSX) custom UI footer. For example, you can use
 #### `viewModeEnabled`
 
 This prop indicates if the app is in `view mode`. When this prop is used, the `view mode` will not show up in context menu is so it is fully controlled by host. Also the value of this prop if passed will be used over the value of `intialData.appState.viewModeEnabled`
+
+#### `zenModeEnabled`
+
+This prop indicates if zen mode is enabled. When this prop is used, the `zen mode` will not show up in context menu is so it is fully controlled by host. The `Exit zen mode` button will also not show up if this prop is true. Also the value of this prop if passed will be used over the value of `intialData.appState.zenModeEnabled`
```
Comment (ad1992) [inline src/packages/excalidraw/README.md:343]:
```suggestion
This prop indicates if` zen mode` is enabled. When this prop is used, the `zen mode` will not show up in the context menu is so it is fully controlled by the host. The `Exit zen mode` button will also not show up if this prop is true. Also, the value of this prop if passed will be used over the value of `intialData.appState.zenModeEnabled`
```

File: src/tests/excalidrawPackage.test.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      });
+      const contextMenu = document.querySelector(".context-menu");
+      fireEvent.click(queryByText(contextMenu as HTMLElement, "Zen mode")!);
+      expect(h.state.zenModeEnabled).toBe(true);
+      expect(
+        container.getElementsByClassName("disable-zen-mode--visible").length,
+      ).toBe(1);
+    });
+
+    it("should not show exit zen mode button and zen mode option in context menu when zenModeEnabled is not undefined", async () => {
```
Comment (ad1992) [inline src/tests/excalidrawPackage.test.tsx:31]:
```suggestion
    it("should not show exit zen mode button and zen mode option in context menu when zenModeEnabled is set", async () => {
```

File: src/packages/excalidraw/README.md [hunk 4]
```diff
@@ -335,3 +337,11 @@ A function that renders (returns JSX) custom UI footer. For example, you can use
 #### `viewModeEnabled`
 
 This prop indicates if the app is in `view mode`. When this prop is used, the `view mode` will not show up in context menu is so it is fully controlled by host. Also the value of this prop if passed will be used over the value of `intialData.appState.viewModeEnabled`
+
+#### `zenModeEnabled`
+
+This prop indicates if` zen mode` is enabled. When this prop is used, the `zen mode` will not show up in the context menu is so it is fully controlled by the host. The `Exit zen mode` button will also not show up if this prop is true. Also, the value of this prop if passed will be used over the value of `intialData.appState.zenModeEnabled`
```
Comment (lipis) [inline src/packages/excalidraw/README.md:343]:
```suggestion
This prop indicates if `zen mode` is enabled. When this prop is used, the `zen mode` will not show up in the context menu is so it is fully controlled by the host. The `Exit zen mode` button will also not show up if this prop is true. Also, the value of this prop if passed will be used over the value of `intialData.appState.zenModeEnabled`
```

**Comment by ad1992** [comment]
> Maybe write a couple of tests coz I have a feeling this will regress :)

@dwelle done 👍

File: src/packages/excalidraw/README.md [hunk 5]
```diff
@@ -335,3 +337,11 @@ A function that renders (returns JSX) custom UI footer. For example, you can use
...
 
 This prop indicates if the app is in `view mode`. When this prop is used, the `view mode` will not show up in context menu is so it is fully controlled by host. Also the value of this prop if passed will be used over the value of `intialData.appState.viewModeEnabled`
+
+#### `zenModeEnabled`
+
+This prop indicates if `zen mode` is enabled. When this prop is used, the `zen mode` will not show up in the context menu is so it is fully controlled by the host. The `Exit zen mode` button will also not show up if this prop is true. Also, the value of this prop if passed will be used over the value of `intialData.appState.zenModeEnabled`
+
+#### `gridModeEnabled`
+
+This prop indicates if `grid mode` is enabled. When this prop is used, the `grid mode` will not show up in the context menu is so it is fully controlled by the host. Also, the value of this prop if passed will be used over the value of `intialData.appState.gridModeEnabled`
```
Comment (dwelle) [inline src/packages/excalidraw/README.md:347]:
```suggestion
#### `viewModeEnabled`

This prop indicates whether the app is in `view mode`. When supplied, the value takes precedence over `intialData.appState.viewModeEnabled`, the `view mode` will be fully controlled by the host app, and users won't be able to toggle it from within the app.

#### `zenModeEnabled`

This prop indicates whether the app is in `zen mode`. When supplied, the value takes precedence over `intialData.appState.zenModeEnabled`, the `zen mode` will be fully controlled by the host app, and users won't be able to toggle it from within the app.

#### `gridModeEnabled`

This prop indicates whether the shows the grid. When supplied, the value takes precedence over `intialData.appState.gridModeEnabled`, the grid will be fully controlled by the host app, and users won't be able to toggle it from within the app.
```

File: src/packages/excalidraw/CHANGELOG.md [hunk 6]
```diff
@@ -18,6 +18,7 @@ Please add the latest change on the top under the correct section.
 
 ### Features
 
+- Add `zenModeEnabled` and `gridModeEnabled` prop which enables zen mode and grid mode respectively [#2901](https://github.com/excalidraw/excalidraw/pull/2901). When this prop is used, the zen mode / grid mode will not show up in context menu is so it is fully controlled by host.
```
Comment (dwelle) [inline src/packages/excalidraw/CHANGELOG.md:21]:
```suggestion
- Add `zenModeEnabled` and `gridModeEnabled` prop which enables zen mode and grid mode respectively [#2901](https://github.com/excalidraw/excalidraw/pull/2901). When this prop is used, the zen mode / grid mode will be fully controlled by the host app.
```

**Review by dwelle — APPROVED** [review]

**Pull request merged**